packages/shared/src/hooks/useExtensionPermission.ts

```diff
@@ -0,0 +1,38 @@
+import { useQueryClient } from 'react-query';
+
+export const EXTENSION_PERMISSION_KEY = 'extension-permission';
+
+export type RequestContentScripts = (
+  origin: string,
+  onPermission?: UseExtensionPermissionProps['onPermission'],
+) => () => Promise<boolean>;
+
+interface ExtensionPermission {
+  isFetched?: boolean;
+  contentScriptGranted: boolean;
+  requestContentScripts: () => Promise<boolean>;
+}
+
+interface UseExtensionPermissionProps {
+  origin: string;
+  onPermission?: (granted: boolean) => void;
+}
+
+export const useExtensionPermission = ({
+  origin,
+  onPermission,
+}: UseExtensionPermissionProps): ExtensionPermission => {
+  const client = useQueryClient();
+  const { requestContentScripts, contentScriptGranted, isFetched } =
+    client.getQueryData<{
+      requestContentScripts: RequestContentScripts;
+      contentScriptGranted: boolean;
+      isFetched: boolean;
+    }>(EXTENSION_PERMISSION_KEY) || {};
+
+  return {
```

Comment:
Not sure we need this one or the other one.

Reply:
This one is used when we need to use this hook inside a component that is shared between the webapp and extension packages. Long term, we would remove both and have a dedicated context handling this.